src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/Helpers.cs

```diff
@@ -118,6 +118,32 @@ public static byte[] FixupKeyParity(this byte[] key)
             }
             return oddParityKey;
         }
+
+        public static void ConvertIntToByteArray(uint dwInput, ref byte[] counter)
```

Comment:
> Why is counter being passed by ref?

Because it is in desktop!  :smile:  The loop that calls this would do it ~12 times (RSA-2048 size / SHA-1 size); so it saves 11 array allocations by just writing into the same one over and over.